homeassistant/components/vicare/climate.py

```diff
@@ -111,7 +103,7 @@ async def async_setup_entry(
     entities = []
     api = hass.data[DOMAIN][config_entry.entry_id][VICARE_API]
     device_config = hass.data[DOMAIN][config_entry.entry_id][VICARE_DEVICE_CONFIG]
-    circuits = await hass.async_add_executor_job(_get_circuits, api)
+    circuits = get_circuits(api)
```

Comment:
This used to run in the executor. Isn't that needed anymore? Ie, is this async safe?

Reply:
In most places, the circuits were accessed without the executor. I had no issues with removing that here as well.

Reply:
Please look in the library what it does. It's either safe or not safe to run in the event loop.

Reply:
I looked in the library and either this does a request or if there's a cache it acquires a lock. Both are not async safe, so we need to run this in the executor.

Reply:
Sorry I missed that.

I looked at get_circuits and thought it was just returning a variable.  But it has a property getter which makes an API call, and underneath, indeed can get data from a URL.

Note that the same is true for `get_compressors` and `get_burners`, so even though they weren't using the executor before the PR, they probably need to.  Happy to help fix this; I don't have a physical system to test it on though.

Reply:
Thanks for checking that, @MartinHjelmare. Will create a PR to fix this for all the helpers.

Reply:
There could be more issues.  

If understand correctly climate.py:`self._circuit.setMode(vicare_mode)` is also doing a request, so I think should be in the executor too (please tell me if I'm mistaken!).

Reply:
`set_hvac_mode` is already run in the executor.

Reply:
I've added a PR here: https://github.com/home-assistant/core/pull/104541